clock/test_clock.go

```diff
@@ -32,7 +46,14 @@ func (c *TestClock) Now() time.Time {
 // duration has passed passed by the user set test time.
 func (c *TestClock) TickAfter(duration time.Duration) <-chan time.Time {
 	c.timeLock.Lock()
-	defer c.timeLock.Unlock()
+	defer func() {
+		c.timeLock.Unlock()
+
+		// Signal that the ticker has been added.
+		if c.tickSignal != nil {
+			c.tickSignal <- struct{}{}
```

Comment:
Could be slightly more useful if the duration is passed in instead of an empty struct (nb)

Reply:
Definitely an option to consider, but imho it is more consistent if `TickAfter()` has the same time source as `Now()`.

Reply:
That wouldn't change the passed in duration

Reply:
Nice thing is also that you can assert that the unit that is being tested requested a timer for the expected duration

Reply:
exactly

Reply:
What do you mean by 'exactly'? That you'd like to change the channel type? Or do it another time?

Reply:
Sorry, misunderstood your comment. I thought you meant I should inject the tick itself which would be a separate time source.
Should be fixed according to your suggestion now.